src/workflows/registration.ts

```diff
@@ -54,14 +42,25 @@ export async function registerOffchainWorkflow({
     },
   });
 
-  return;
+  return registeredUser.data;
+}
+
+interface IsRegisteredCheckError {
+  reason: string;
 }
 
 export async function isRegisteredOnChainWorkflow(
   starkPublicKey: string,
   contract: Registration,
 ): Promise<boolean> {
-  return await contract.isRegistered(starkPublicKey);
+  try {
+    return await contract.isRegistered(starkPublicKey);
+  } catch (ex) {
+    if ((ex as IsRegisteredCheckError).reason === 'USER_UNREGISTERED') {
+      return false;
+    }
+    throw ex;
+  }
```

Comment:
cleaner suggestion:

```suggestion
  try {
    return await contract.isRegistered(starkPublicKey);
  } catch (err) {
    if (err.reason === 'USER_UNREGISTERED') return false;
    throw ex;
  }
```

Reply:
Typescript doesn't like that. It wouldn't complain if we did `catch (err: any)` though.

Reply:
ah makes sense. fair enough